src/ext/command.rs

```diff
@@ -1,16 +1,512 @@
+    /// then the input:
+    /// ```ignore
+    /// Value::new_secret(
+    ///     "AUTHORIZATION: Basic username:password",
+    ///     "AUTHORIZATION: Basic {secret}",
+    /// )
+    /// ```
+    /// Is output as:
+    /// ```not_rust
+    /// AUTHORIZATION: Basic <REDACTED>
+    /// ```
+    ///
+    /// If there are _no_ `{secret}` literals, the string is used as-is.
+    /// If there are _multiple_ `{secret}` literals, they are all replaced.
+    ///
+    /// The literal actually used at runtime is managed by [`REDACTION_LITERAL`].
+    ///
+    /// # Background
+    ///
+    /// Generally intended to be used when an entire string is provided as a value, but only part of it is actually secret,
+    /// to support showing as much as possible of the input.
+    ///
+    /// For example, the git arguments:
+    /// ```not_rust
+    /// git -c 'http.extraHeader=AUTHORIZATION: Basic username:password'
+    /// ```
+    ///
+    /// Would be have to be passed in as:
+    /// ```ignore
+    /// vec![Value::new_plain("-c"), Value::new_secret("http.extraHeader=AUTHORIZATION: Basic username:password")]
+    /// ```
+    ///
+    /// Which would result in less than helpful output, since the entire secret output is redacted:
+    /// ```not_rust
+    /// run git: /some/path/to/git
+    /// args: ["-c", "<REDACTED>"]
+    /// ```
+    ///
+    /// But using this variant:
+    /// ```ignore
+    /// vec![
+    ///     Value::new_plain("-c"),
+    ///     Value::new_secret_format(
+    ///         "http.extraHeader=AUTHORIZATION: Basic username:password",
+    ///         "http.extraHeader=AUTHORIZATION: Basic {}"
+    ///     ),
+    /// ]
+    /// ```
+    ///
+    /// We can get back more useful debugging output:
+    /// ```ignore
+    /// run git: /some/path/to/git
+    /// args: ["-c", "http.extraHeader=AUTHORIZATION: Basic <REDACTED>"]
+    /// ```
+    ///
+    /// Note that this only applies to debug output generated by Broker;
+    /// redactions applied to output generated by the child process
+    /// uses the simpler initial form.
+    SecretFormat(ComparableSecretString, String),
+
+    /// Plain text, not redacted in debugging output.
+    Plain(String),
+}
+
+impl Value {
+    /// Create a new instance as a `Secret` variant.
+    ///
+    /// `Secret` variants are automatically redacted in debugging output.
+    pub fn new_secret<S: Into<ComparableSecretString>>(value: S) -> Self {
+        Self::Secret(value.into())
+    }
+
+    /// Create a new instance as a `Secret` variant, with a format to use when printing the secret.
+    ///
+    /// `Secret` variants are automatically redacted in debugging output,
+    /// but this version applies the redacted string to the provided format string.
+    /// See [`Value::SecretFormat`] for more details.
+    pub fn new_secret_format<S, F>(value: S, format: F) -> Self
+    where
+        S: Into<ComparableSecretString>,
+        F: Into<String>,
+    {
```

Comment:
I feel like these args are in the opposite order from all of the other `format!` commands. I.e. usually the format string comes first.

But maybe there are non-macro variants that I'm not familiar with?

Reply:
Oh, that's a really good point! I've updated this to be more format-like.